src/zcash/JoinSplit.cpp

```diff
@@ -1,213 +1,212 @@
 #include "JoinSplit.hpp"
-#include "prf.h"
-
-#include "zcash/util.h"
-
-#include <memory>
 
-#include <boost/format.hpp>
-#include <fstream>
-#include "tinyformat.h"
-#include "sync.h"
 #include "amount.h"
-
 #include "librustzcash.h"
+#include "prf.h"
 #include "streams.h"
+#include "sync.h"
+#include "tinyformat.h"
 #include "version.h"
+#include "zcash/util.h"
 
+#include <boost/format.hpp>
 #include <rust/blake2b.h>
 
-namespace libzcash {
-
-    template<size_t NumInputs, size_t NumOutputs>
-    SproutProof JoinSplit<NumInputs, NumOutputs>::prove(
-        const std::array<JSInput, NumInputs>& inputs,
-        const std::array<JSOutput, NumOutputs>& outputs,
-        std::array<SproutNote, NumOutputs>& out_notes,
-        std::array<ZCNoteEncryption::Ciphertext, NumOutputs>& out_ciphertexts,
-        uint256& out_ephemeralKey,
-        const ed25519::VerificationKey& joinSplitPubKey,
-        uint256& out_randomSeed,
-        std::array<uint256, NumInputs>& out_macs,
-        std::array<uint256, NumInputs>& out_nullifiers,
-        std::array<uint256, NumOutputs>& out_commitments,
-        uint64_t vpub_old,
-        uint64_t vpub_new,
-        const uint256& rt,
-        bool computeProof,
-        uint256 *out_esk // Payment disclosure
-    ) {
-        if (vpub_old > MAX_MONEY) {
-            throw std::invalid_argument("nonsensical vpub_old value");
-        }
+#include <fstream>
+#include <memory>
```

Comment:
This is another disadvantage of reordering headers, it can lead to confusing diffs (and we do need to fully review the clang-format changes).